### ISSUE TITLE
OpenShift s2i binary builds seem broken in 6.3

### ISSUE THREAD

**Issue by manusa** [fabric8io/kubernetes-client#4735]
### Describe the bug

While upgrading Eclipse JKube to use Kubernetes Client 6.3.1 https://github.com/eclipse/jkube/pull/1988, we've detected that binary builds are not completed.

It seems that the binary upload is not working as expected and the HTTP request never completes.

The call:
```java
client.buildConfigs()
  .inNamespace(namespace)
  .withName(buildName)
  .instantiateBinary()
  .fromFile(binaryTarFile);
```

will hang forever.

There's a very simple unit test to verify this logic:
https://github.com/fabric8io/kubernetes-client/blob/1e944fc144d514e5a7ef875e0552be039fed75cf/kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/BuildConfigTest.java#L208

But there's no real e2e test verifying this scenario.

### Fabric8 Kubernetes Client version

6.3.1

### Steps to reproduce

Create an OpenShift binary build:
```java
client.buildConfigs()
  .inNamespace(namespace)
  .withName(buildName)
  .instantiateBinary()
  .fromFile(binaryTarFile);
```

### Expected behavior

The binary build should complete successfully

### Runtime

OpenShift

### Kubernetes API Server version

other (please specify in additional context)

### Environment

Linux

### Additional context

Point of failure:
https://github.com/fabric8io/kubernetes-client/blob/1d3e263cc6216f4a3adad6b677539186179f41c2/openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/build/BuildConfigOperationsImpl.java#L267-L271

https://github.com/fabric8io/kubernetes-client/blob/1d3e263cc6216f4a3adad6b677539186179f41c2/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/OperationSupport.java#L515

**Comment by shawkins** [comment]
More than likely the underlying issue is not related to this particular call - it's just a post of string value that is no different than a typical create call.  This in part was to be addressed by #4663 - but it would be great if your scenario is reproducible to get a confirmation of what is occurring.

**Comment by manusa** [comment]
> What client is being used in this scenario?

OkHttp / OpenShiftClient

https://github.com/eclipse/jkube/blob/d7b92219c2fe94d68831e9016b943d851a0040e0/jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/openshift/OpenshiftBuildService.java#L443

When upgrading JKube to 6.3.1 (https://github.com/eclipse/jkube/pull/1988) the oc:build goal which performs an S2i binary build will freeze in the step when the binary is uploaded to the server. Cancelling the local operation will unlock the operation so the server will continue with the build. It's my understanding that the binary is uploaded, but the client doesn't get notified of the request completion (but need to investigate more though)

**Comment by manusa** [comment]
It seems that the [`Expect`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Expect) header is using a wrong value:

https://github.com/fabric8io/kubernetes-client/blob/73572bb8d08459791f8d23fe3fa48a96b8f6f10d/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/http/StandardHttpHeaders.java#L28

This value is case-sensitive and should be `100-continue` instead.

Since `expectContinue` is used here:
https://github.com/fabric8io/kubernetes-client/blob/73572bb8d08459791f8d23fe3fa48a96b8f6f10d/openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/build/BuildConfigOperationsImpl.java#L265

and this is the switch to the constant is the most notable change between 6.1 and 6.3, it's very likely that the problem is related to it.

An easy way to check this is by using the JDK client instead, since it's not relying on the constant.

**Comment by manusa** [comment]
Should be fixed by #4766
